x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java

```diff
@@ -170,7 +170,8 @@ public void testHandshakeFailureBecauseProtocolMismatch() throws Exception {
             // fips JSSE does not support TLSv1.3 yet
             serverProtocols = new String[]{"TLSv1.2"};
             clientProtocols = new String[]{"TLSv1.1"};
-            expectedMessageMatcher = is("org.bouncycastle.tls.TlsFatalAlert: protocol_version(70)");
+            expectedMessageMatcher = anyOf(is("The client supported protocol versions [TLSv1.1] are not accepted by server preferences " +
```

Comment:
How about we ditch the entire `if ( inFipsJvm() )` branch and add  `is("org.bouncycastle.tls.TlsFatalAlert: protocol_version(70)")` to the anyOf of the `else` below? 
Given that this is just the manifestation of the failure we're catching and the specifics of where it is thrown, I don't think we miss anything in coverage

Reply:
I think the `if (inFipsJvm())` is required because the tested protocol mismatch is between TLSv1.1 and TLSv1.2 even in the case where we're running in JDK 16 (JDK 16 doesn't work with TLSv1.1 , unless in FIPS mode). Or maybe I've not understood the suggestion?

Reply:
No, you're right I missed that. We'd need `&& inFipsJvm()` in the `if` branch. Happy to leave this is explicit as is.